### ISSUE TITLE
Frontend Subroute Fails to Correctly Route in Local Development

### ISSUE THREAD

**Issue by sennyeya** [backstage/backstage#14085]
## Expected Behavior

When a user navigates to `http://localhost:3000/{basePath}/`, they should be delivered to `http://localhost:3000/{basePath}/catalog`.

## Actual Behavior

When a user navigates to `http://localhost:3000/{basePath}/`, they are navigated to `http://localhost:3000/catalog` causing the entire webpage to fail to render anything.

## Steps to Reproduce

Testing with this project's `example-app` on `38b94b25adcb022df3e1ecaf07ac43b05c0fb5fe`.

1. Set `app.baseUrl` in app-config.yaml to something with a subroute.
2. Run `yarn dev`, try to navigate to the URL, `http://localhost:3000/{basePath}/`.

## Possible Cause
I think https://github.com/backstage/backstage/blob/master/packages/app/src/App.tsx#L153 is causing the issue, since the `Navigate` component comes directly from react-router and is still treating `http://localhost:3000/` as the base path.

## Context

Attempting to test subroute routing locally, currently unable to test due to this bug and related WebpackDevServer issue https://github.com/backstage/backstage/pull/13927/files#diff-75f058db20a95677f58a9364712e2bf3a5538300330c6fcbb6a5288885e421d8R65.

## Your Environment

- Browser Information: 
Tested in both 
Google Chrome: Version 106.0.5249.103 (Official Build) (arm64)
Firefox: 102.2.0esr (64-bit).

- Output of `yarn backstage-cli info`: 

```text
OS:   Darwin 21.6.0 - darwin/arm64
node: v16.17.0
yarn: 3.2.3
cli:  0.20.0-next.2 (local)
backstage:  N/A

Dependencies:
  @backstage/app-defaults                                   0.0.0-use.local
  @backstage/backend-app-api                                0.0.0-use.local
  @backstage/backend-common                                 0.0.0-use.local
  @backstage/backend-defaults                               0.0.0-use.local
  @backstage/backend-plugin-api                             0.0.0-use.local
  @backstage/backend-tasks                                  0.0.0-use.local
  @backstage/backend-test-utils                             0.0.0-use.local
  @backstage/catalog-client                                 1.1.0, 0.0.0-use.local
  @backstage/catalog-model                                  1.1.1, 0.0.0-use.local
  @backstage/cli-common                                     0.0.0-use.local
  @backstage/cli                                            0.0.0-use.local
  @backstage/codemods                                       0.0.0-use.local
  @backstage/config-loader                                  0.0.0-use.local
  @backstage/config                                         1.0.2, 0.0.0-use.local
  @backstage/core-app-api                                   0.0.0-use.local
  @backstage/core-components                                0.11.1, 0.0.0-use.local
  @backstage/core-plugin-api                                1.0.6, 0.0.0-use.local
  @backstage/create-app                                     0.0.0-use.local
  @backstage/dev-utils                                      0.0.0-use.local
  @backstage/errors                                         1.1.1, 0.0.0-use.local
  @backstage/integration-react                              1.1.4, 0.0.0-use.local
  @backstage/integration                                    1.3.1, 0.0.0-use.local
  @backstage/plugin-adr-backend                             0.0.0-use.local
  @backstage/plugin-adr-common                              0.0.0-use.local
  @backstage/plugin-adr                                     0.0.0-use.local
  @backstage/plugin-airbrake-backend                        0.0.0-use.local
  @backstage/plugin-airbrake                                0.0.0-use.local
  @backstage/plugin-allure                                  0.0.0-use.local
  @backstage/plugin-analytics-module-ga                     0.0.0-use.local
  @backstage/plugin-apache-airflow                          0.0.0-use.local
  @backstage/plugin-api-docs-module-protoc-gen-doc          0.0.0-use.local
  @backstage/plugin-api-docs                                0.0.0-use.local
  @backstage/plugin-apollo-explorer                         0.0.0-use.local
  @backstage/plugin-app-backend                             0.0.0-use.local
  @backstage/plugin-auth-backend                            0.0.0-use.local
  @backstage/plugin-auth-node                               0.0.0-use.local
  @backstage/plugin-azure-devops-backend                    0.0.0-use.local
  @backstage/plugin-azure-devops-common                     0.0.0-use.local
  @backstage/plugin-azure-devops                            0.0.0-use.local
  @backstage/plugin-badges-backend                          0.0.0-use.local
  @backstage/plugin-badges                                  0.0.0-use.local
  @backstage/plugin-bazaar-backend                          0.0.0-use.local
  @backstage/plugin-bazaar                                  0.0.0-use.local
  @backstage/plugin-bitbucket-cloud-common                  0.0.0-use.local
  @backstage/plugin-bitrise                                 0.0.0-use.local
  @backstage/plugin-catalog-backend-module-aws              0.0.0-use.local
  @backstage/plugin-catalog-backend-module-azure            0.0.0-use.local
  @backstage/plugin-catalog-backend-module-bitbucket-cloud  0.0.0-use.local
  @backstage/plugin-catalog-backend-module-bitbucket-server 0.0.0-use.local
  @backstage/plugin-catalog-backend-module-bitbucket        0.0.0-use.local
  @backstage/plugin-catalog-backend-module-gerrit           0.0.0-use.local
  @backstage/plugin-catalog-backend-module-github           0.0.0-use.local
  @backstage/plugin-catalog-backend-module-gitlab           0.0.0-use.local
  @backstage/plugin-catalog-backend-module-ldap             0.0.0-use.local
  @backstage/plugin-catalog-backend-module-msgraph          0.0.0-use.local
  @backstage/plugin-catalog-backend-module-openapi          0.0.0-use.local
  @backstage/plugin-catalog-backend                         0.0.0-use.local
  @backstage/plugin-catalog-common                          1.0.6, 0.0.0-use.local
  @backstage/plugin-catalog-graph                           0.0.0-use.local
  @backstage/plugin-catalog-graphql                         0.0.0-use.local
  @backstage/plugin-catalog-import                          0.0.0-use.local
  @backstage/plugin-catalog-node                            0.0.0-use.local
  @backstage/plugin-catalog-react                           1.1.4, 0.0.0-use.local
  @backstage/plugin-catalog                                 0.0.0-use.local
  @backstage/plugin-cicd-statistics-module-gitlab           0.0.0-use.local
  @backstage/plugin-cicd-statistics                         0.0.0-use.local
  @backstage/plugin-circleci                                0.0.0-use.local
  @backstage/plugin-cloudbuild                              0.0.0-use.local
  @backstage/plugin-code-climate                            0.0.0-use.local
  @backstage/plugin-code-coverage-backend                   0.0.0-use.local
  @backstage/plugin-code-coverage                           0.0.0-use.local
  @backstage/plugin-codescene                               0.0.0-use.local
  @backstage/plugin-config-schema                           0.0.0-use.local
  @backstage/plugin-cost-insights-common                    0.0.0-use.local
  @backstage/plugin-cost-insights                           0.0.0-use.local
  @backstage/plugin-dynatrace                               0.0.0-use.local
  @backstage/plugin-explore-react                           0.0.0-use.local
  @backstage/plugin-explore                                 0.0.0-use.local
  @backstage/plugin-firehydrant                             0.0.0-use.local
  @backstage/plugin-fossa                                   0.0.0-use.local
  @backstage/plugin-gcalendar                               0.0.0-use.local
  @backstage/plugin-gcp-projects                            0.0.0-use.local
  @backstage/plugin-git-release-manager                     0.0.0-use.local
  @backstage/plugin-github-actions                          0.0.0-use.local
  @backstage/plugin-github-deployments                      0.0.0-use.local
  @backstage/plugin-github-issues                           0.0.0-use.local
  @backstage/plugin-github-pull-requests-board              0.0.0-use.local
  @backstage/plugin-gitops-profiles                         0.0.0-use.local
  @backstage/plugin-gocd                                    0.0.0-use.local
  @backstage/plugin-graphiql                                0.0.0-use.local
  @backstage/plugin-graphql-backend                         0.0.0-use.local
  @backstage/plugin-home                                    0.4.25, 0.0.0-use.local
  @backstage/plugin-ilert                                   0.0.0-use.local
  @backstage/plugin-jenkins-backend                         0.0.0-use.local
  @backstage/plugin-jenkins-common                          0.0.0-use.local
  @backstage/plugin-jenkins                                 0.0.0-use.local
  @backstage/plugin-kafka-backend                           0.0.0-use.local
  @backstage/plugin-kafka                                   0.0.0-use.local
  @backstage/plugin-kubernetes-backend                      0.0.0-use.local
  @backstage/plugin-kubernetes-common                       0.0.0-use.local
  @backstage/plugin-kubernetes                              0.0.0-use.local
  @backstage/plugin-lighthouse                              0.0.0-use.local
  @backstage/plugin-newrelic-dashboard                      0.0.0-use.local
  @backstage/plugin-newrelic                                0.0.0-use.local
  @backstage/plugin-org                                     0.0.0-use.local
  @backstage/plugin-pagerduty                               0.0.0-use.local
  @backstage/plugin-periskop-backend                        0.0.0-use.local
  @backstage/plugin-periskop                                0.0.0-use.local
  @backstage/plugin-permission-backend                      0.0.0-use.local
  @backstage/plugin-permission-common                       0.6.4, 0.0.0-use.local
  @backstage/plugin-permission-node                         0.0.0-use.local
  @backstage/plugin-permission-react                        0.4.5, 0.0.0-use.local
  @backstage/plugin-playlist-backend                        0.0.0-use.local
  @backstage/plugin-playlist-common                         0.0.0-use.local
  @backstage/plugin-playlist                                0.0.0-use.local
  @backstage/plugin-proxy-backend                           0.0.0-use.local
  @backstage/plugin-rollbar-backend                         0.0.0-use.local
  @backstage/plugin-rollbar                                 0.0.0-use.local
  @backstage/plugin-scaffolder-backend-module-cookiecutter  0.0.0-use.local
  @backstage/plugin-scaffolder-backend-module-rails         0.0.0-use.local
  @backstage/plugin-scaffolder-backend-module-yeoman        0.0.0-use.local
  @backstage/plugin-scaffolder-backend                      0.0.0-use.local
  @backstage/plugin-scaffolder-common                       0.0.0-use.local
  @backstage/plugin-scaffolder                              0.0.0-use.local
  @backstage/plugin-search-backend-module-elasticsearch     0.0.0-use.local
  @backstage/plugin-search-backend-module-pg                0.0.0-use.local
  @backstage/plugin-search-backend-node                     0.0.0-use.local
  @backstage/plugin-search-backend                          0.0.0-use.local
  @backstage/plugin-search-common                           1.0.1, 0.0.0-use.local
  @backstage/plugin-search-react                            0.0.0-use.local
  @backstage/plugin-search                                  0.0.0-use.local
  @backstage/plugin-sentry                                  0.0.0-use.local
  @backstage/plugin-shortcuts                               0.0.0-use.local
  @backstage/plugin-sonarqube-backend                       0.0.0-use.local
  @backstage/plugin-sonarqube                               0.0.0-use.local
  @backstage/plugin-splunk-on-call                          0.0.0-use.local
  @backstage/plugin-stack-overflow-backend                  0.0.0-use.local
  @backstage/plugin-stack-overflow                          0.1.5, 0.0.0-use.local
  @backstage/plugin-tech-insights-backend-module-jsonfc     0.0.0-use.local
  @backstage/plugin-tech-insights-backend                   0.0.0-use.local
  @backstage/plugin-tech-insights-common                    0.0.0-use.local
  @backstage/plugin-tech-insights-node                      0.0.0-use.local
  @backstage/plugin-tech-insights                           0.0.0-use.local
  @backstage/plugin-tech-radar                              0.0.0-use.local
  @backstage/plugin-techdocs-addons-test-utils              0.0.0-use.local
  @backstage/plugin-techdocs-backend                        0.0.0-use.local
  @backstage/plugin-techdocs-module-addons-contrib          0.0.0-use.local
  @backstage/plugin-techdocs-node                           0.0.0-use.local
  @backstage/plugin-techdocs-react                          0.0.0-use.local
  @backstage/plugin-techdocs                                0.0.0-use.local
  @backstage/plugin-todo-backend                            0.0.0-use.local
  @backstage/plugin-todo                                    0.0.0-use.local
  @backstage/plugin-user-settings-backend                   0.0.0-use.local
  @backstage/plugin-user-settings                           0.0.0-use.local
  @backstage/plugin-vault-backend                           0.0.0-use.local
  @backstage/plugin-vault                                   0.0.0-use.local
  @backstage/plugin-xcmetrics                               0.0.0-use.local
  @backstage/release-manifests                              0.0.0-use.local
  @backstage/test-utils                                     0.0.0-use.local
  @backstage/theme                                          0.0.0-use.local
  @backstage/types                                          0.0.0-use.local
  @backstage/version-bridge                                 0.0.0-use.local
```

**Comment by Rugvip** [comment]
What does your `<Navigate to=... />` look like? It should be `to="catalog"`, not an absolute path. That's for it to work with both react-router beta and stable anyway. Now that react-router has proper support for basename in the router I think we'll add that too though, meaning `to="/catalog"` would work as well, although that might need some more work.

**Comment by Rugvip** [comment]
So `to="catalog"` is the immediate fix right now, but I'm also proposing #14165 to fix the broader issue of base paths being a bit wonky still. A reason it's been so hard to get it working well so far has been that React Router v6 didn't have it built-in, but now that they do we should rely on that implementation instead.

**Comment by Rugvip** [comment]
Oh we were just about to release that apparently 😅, I was testing with 1.6

Fixed in #14187
